zcash_client_backend/src/data_api/testing/pool.rs

```diff
@@ -86,6 +86,11 @@ use zcash_protocol::PoolType;
 #[cfg(feature = "pczt")]
 use pczt::roles::{prover::Prover, signer::Signer};
 
+/// The number of ephemeral addresses that can be safely reserved without observing any
+/// of them to be mined.
+#[cfg(feature = "transparent-inputs")]
+pub const EPHEMERAL_ADDR_GAP_LIMIT: usize = 3;
```

Comment:
This is quite aggressively low. The issue with a low gap limit is that there might be situations in which the wallet could request an address and then, because of some temporary condition, not be able to create or submit the transaction (i.e. it will never be mined). If it does that 3 times in succession then the wallet will be bricked for ephemeral address usage.